daemon/src/position_metrics.rs

```diff
@@ -0,0 +1,344 @@
+            id,
+            position,
+            quantity_usd,
+
+            is_open: false,
+            is_closed: true,
+            is_failed: false,
+            is_refunded: false,
+            version: 0,
+        }
+    }
+}
+
+mod metrics {
+    use crate::position_metrics::Cfd;
+    use model::Position;
+    use model::Usd;
+    use rust_decimal::prelude::ToPrimitive;
+    use std::collections::HashMap;
+    use std::iter::Filter;
+    use std::slice::Iter;
+
+    const POSITION_LABEL: &str = "position";
+    const POSITION_LONG_LABEL: &str = "long";
+    const POSITION_SHORT_LABEL: &str = "short";
+
+    const STATUS_LABEL: &str = "status";
+    const STATUS_OPEN_LABEL: &str = "open";
+    const STATUS_CLOSED_LABEL: &str = "closed";
+    const STATUS_FAILED_LABEL: &str = "failed";
+    const STATUS_REFUNDED_LABEL: &str = "refunded";
+    // this is needed so that we do not lose cfds which are in a weird state, e.g. open & closed.
+    // This should be 0 though but for the time being we add it
+    const STATUS_UNKNOWN_LABEL: &str = "unknown";
+
+    static POSITION_QUANTITY_GAUGE: conquer_once::Lazy<prometheus::GaugeVec> =
+        conquer_once::Lazy::new(|| {
+            prometheus::register_gauge_vec!(
+                "positions_quantity",
+                "Total quantity of positions on ItchySats.",
+                &[POSITION_LABEL, STATUS_LABEL]
+            )
+            .unwrap()
+        });
+
+    static POSITION_AMOUNT_GAUGE: conquer_once::Lazy<prometheus::IntGaugeVec> =
+        conquer_once::Lazy::new(|| {
+            prometheus::register_int_gauge_vec!(
+                "positions_total",
+                "Total number of positions on ItchySats.",
+                &[POSITION_LABEL, STATUS_LABEL]
+            )
+            .unwrap()
+        });
+
+    pub fn update_position_metrics(cfds: &[Cfd]) {
+        set_position_metrics(cfds.iter().filter(|cfd| cfd.is_open), STATUS_OPEN_LABEL);
+        set_position_metrics(cfds.iter().filter(|cfd| cfd.is_closed), STATUS_CLOSED_LABEL);
+        set_position_metrics(cfds.iter().filter(|cfd| cfd.is_failed), STATUS_FAILED_LABEL);
+        set_position_metrics(
+            cfds.iter().filter(|cfd| cfd.is_refunded),
+            STATUS_REFUNDED_LABEL,
+        );
+
+        set_position_metrics(
+            cfds.iter().filter(|cfd| {
+                let unknown_state = !(cfd.is_open ^ cfd.is_closed ^ cfd.is_refunded);
+                if unknown_state {
+                    tracing::error!(
+                        is_open = cfd.is_open,
+                        is_closed = cfd.is_closed,
+                        is_refunded = cfd.is_refunded,
+                        order_id = %cfd.id,
+                        "CFD is in weird state"
+                    );
+                }
+                unknown_state
+            }),
+            STATUS_UNKNOWN_LABEL,
+        );
+    }
+
+    fn set_position_metrics(cfds: Filter<Iter<Cfd>, fn(&&Cfd) -> bool>, status: &str) {
```

Comment:
```suggestion
    fn set_position_metrics(cfds: impl Iterator<Item = &Cfd>, status: &str) {
```

This should do the trick as well.